### DISCUSSION TITLE
Support log in and decrypt with passkeys

### DISCUSSION THREAD

**Discussion by septatrix** [dani-garcia/vaultwarden#4250] [Ideas]
A week ago Bitwarden added support to log in and decrypt the vault using passkeys/WebAuthn. Supporting this would be great for those who prefer security keys etc. over typing a master password every time.

I am not sure if all changes are contained within the clients and updating the web frontend is sufficient or if this requires new endpoints on the backend. I fear the latter might be the case as the process requires the relying party to provide a salt if I understand correctly which the key uses to derive the unlocking phrase.

**Comment by BlackDex** [comment]
As mentioned in other posts, we need to add new endpoints to get this supported. So only updating the clients will unfortunately not provide this feature.

**Reply by septatrix** [reply]
Hm I searched this repo but did not find anything about it, is this already tracked somewhere (else)?

**Reply by septatrix** [reply]
Or maybe add this to #246?

**Reply by septatrix** [reply]
Do you happen to know which endpoints are missing specifically? How complex are they?

**Reply by BlackDex** [reply]
Besides the endpoints we also need to update some libraries. Which is abit harder. So it's not an easy task, or simply putting something in a database and be done with it. If that was only so easy haha.

**Reply by septatrix** [reply]
Is there a roadmap or any estimate when this might get worked on?

**Reply by septatrix** [reply]
Do you mind adding this to the pinned tracking issue?

**Comment by rbability** [comment]
I just found this one after going through the Changelog of 1.30.2 and saw that it integrated WebVault 2024.1.2, which has Passkey Logon on the official Bitwarden. I already suspected there is more to that than just integrating the current WebVault from Bitwarden. I would really like to see this too here, but I understand that it might be a lot of work, so of course I´m not demanding anything, just telling that there is interest in this feature :-)
Thanks for the great work you are all doing on this Project!

**Comment by tanpro260196** [comment]
Now that Firefox also support PRF extension, the only missing link is Vaultwarden support.\
It would be really nice to have a completely passwordless work flow.

**Reply by Be-ing** [reply]
Unfortunately more work is still needed on Firefox's side to enable the browser extension to unlock with a passkey: https://bugzilla.mozilla.org/show_bug.cgi?id=1956484

**Comment by BlauToad** [comment]
Please add this <3 I haven't used Rust that much, I need to learn Rust just to develop this 😅

**Comment by dominicf2001** [comment]
I would love this feature

**Comment by individ2016** [comment]
I would like this feature too.. What the point to have password storage and to have one password to see all passwords i have..

**Comment by Willywang8216** [comment]
This is a very helpful feature. 🥺🥺🥺

**Comment by Camis** [comment]
https://community.bitwarden.com/t/you-can-now-unlock-your-vault-with-a-passkey/93556

**Comment by a15355447898a** [comment]
I've tried adding this feature, feel free to give it a go (data security is not guaranteed, please back up your data before use).

https://github.com/a15355447898a/vaultwarden/releases/tag/1.0

**Reply by individ2016** [reply]
Maybe you could do pull request for this feature to add it in main brunch?)

**Reply by BlackDex** [reply]
There already is a PR for that, but not yet merged, and it needs some TLC now https://github.com/dani-garcia/vaultwarden/pull/5929

**Comment by SesioN** [comment]
Is this getting reviewed and merged (anyone working on it?)
